src/main/java/in/projecteka/gateway/common/Validator.java

```diff
@@ -27,24 +26,30 @@
     CacheAdapter<String, String> requestIdMappings;
 
 
-    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity) {
-        List<String> xHipIds = requestEntity.getHeaders().get(X_HIP_ID);
-        if (xHipIds == null || xHipIds.isEmpty()) {
-            return Mono.error(ClientError.hipIdMissing());
+    public Mono<ValidatedRequest> validateRequest(HttpEntity<String> requestEntity, String id) {
+        List<String> ids = requestEntity.getHeaders().get(id);
+        if (ids == null || ids.isEmpty()) {
```

Comment:
you can use this here.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/util/CollectionUtils.html#CollectionUtils-